internal/storage/pci/pci.go

```diff
@@ -0,0 +1,84 @@
+package pci
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+// WaitForDevice waits for bus location path of the device to be present
+func WaitForDevice(ctx context.Context, vmBusGUID string) error {
+	_, err := FindDeviceBusLocationFromVMBusGUID(ctx, vmBusGUID)
+	return err
+}
+
+// FindDeviceBusLocationFromVMBusGUID finds device bus location by
+// reading /sys/bus/vmbus/devices/<vmBusGUID>/...
+func FindDeviceBusLocationFromVMBusGUID(ctx context.Context, vmBusGUID string) (string, error) {
+	pciDir, err := findVMBusPCIDirs(ctx, vmBusGUID)
+	if err != nil {
+		return "", err
+	}
+
+	pciDeviceLocation, err := findVMBusPCIDevice(ctx, pciDir)
+	if err != nil {
+		return "", err
+	}
+	return pciDeviceLocation, nil
+}
+
+// findVMBusPCIDevice waits for the pci bus location directory under the path
+// returned from findVMBusPCIDirs to exist and returns the full pci bus location or an error
+func findVMBusPCIDevice(ctx context.Context, pciDirFullPath string) (string, error) {

```

Comment:
I think a comment in here explaining the logic of the paths you are examining would be good. Especially if it includes an example of the paths so we can follow along.

Reply:
Do the new comments help?

Reply:
do the new comments make it clearer?